File: CS/WebUserControl.ascx.cs
using DevExpress.Web;
using DevExpress.Web.Internal;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

public enum CustomControlType { ASPxComboBox, ASPxMenu }
public enum RequestMethod { GET, POST }
public partial class WebUserControl : System.Web.UI.UserControl {

    protected override void OnInit(EventArgs e) {
        base.OnInit(e);
        this.PopulateItems();
    }

    protected override void CreateChildControls() {
        base.CreateChildControls();
        this.PopulateControl();
    }

    Control control;
    public void PopulateControl() {
        if(control != null) this.Controls.Remove(control);
        if(ControlType == CustomControlType.ASPxComboBox) {
            control = this.CreateASPxComboBox();

        } else {
            control = this.CreateASPxMenu();
        }
    }

    private void Pass_CustomJSProperties(object sender, CustomJSPropertiesEventArgs e) {
        e.Properties["cp_RequestMethod"] = RequestMethod;
    }

    public ASPxMenu CreateASPxMenu() {
        ASPxMenu menu = new ASPxMenu();
        menu.ID = "menu";
        menu.AllowSelectItem = true;
        menu.CustomJSProperties += Pass_CustomJSProperties;
        this.Controls.Add(menu);
        menu.EnableClientSideAPI = true;
        menu.ClientSideEvents.SetEventHandler("ItemClick", "OnItemClick");
        menu.ClientSideEvents.SetEventHandler("Init", "OnMenuInit");
        DevExpress.Web.MenuItem item = menu.Items.Add("Select a theme");        
        for(int i = 0; i < Items.Count; i++) {
            item.Items.Add(Items[i].ToString(), Items[i].ToString());
        }
        return menu;
    }

    public ASPxComboBox CreateASPxComboBox() {
        ASPxComboBox combo = new ASPxComboBox();
        combo.ID = "combo";
        combo.CustomJSProperties += Pass_CustomJSProperties;
        this.Controls.Add(combo);
        for(int i = 0; i < Items.Count; i++) {
            combo.Items.Add(Items[i]);
        }
        combo.EnableClientSideAPI = true;
        combo.ClientSideEvents.SetEventHandler("SelectedIndexChanged", "OnSelectedIndexChanged");
        combo.ClientSideEvents.SetEventHandler("Init", "OnComboInit");
        return combo;

    }

    [System.ComponentModel.DefaultValue(RequestMethod.POST)]
    public RequestMethod RequestMethod { get; set; }

    [System.ComponentModel.DefaultValue(CustomControlType.ASPxComboBox)]
    public CustomControlType ControlType { get; set; }

    protected List<string> Items;
    public void PopulateItems() {
        Items = ThemesProvider.GetThemes();
    }

}

File: VB/WebUserControl.ascx.vb
Imports DevExpress.Web
Imports DevExpress.Web.Internal
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls

Public Enum CustomControlType
    ASPxComboBox
    ASPxMenu
End Enum
Public Enum RequestMethod
    [GET]
    POST
End Enum
Partial Public Class WebUserControl
    Inherits System.Web.UI.UserControl

    Protected Overrides Sub OnInit(ByVal e As EventArgs)
        MyBase.OnInit(e)
        Me.PopulateItems()
    End Sub

    Protected Overrides Sub CreateChildControls()
        MyBase.CreateChildControls()
        Me.PopulateControl()
    End Sub

    Private control As Control
    Public Sub PopulateControl()
        If control IsNot Nothing Then
            Me.Controls.Remove(control)
        End If
        If ControlType = CustomControlType.ASPxComboBox Then
            control = Me.CreateASPxComboBox()

        Else
            control = Me.CreateASPxMenu()
        End If
    End Sub

    Private Sub Pass_CustomJSProperties(ByVal sender As Object, ByVal e As CustomJSPropertiesEventArgs)
        e.Properties("cp_RequestMethod") = RequestMethod
    End Sub

    Public Function CreateASPxMenu() As ASPxMenu
        Dim menu As New ASPxMenu()
        menu.ID = "menu"
        menu.AllowSelectItem = True
        AddHandler menu.CustomJSProperties, AddressOf Pass_CustomJSProperties
        Me.Controls.Add(menu)
        menu.EnableClientSideAPI = True
        menu.ClientSideEvents.SetEventHandler("ItemClick", "OnItemClick")
        menu.ClientSideEvents.SetEventHandler("Init", "OnMenuInit")
        Dim item As DevExpress.Web.MenuItem = menu.Items.Add("Select a theme")
        For i As Integer = 0 To Items.Count - 1
            item.Items.Add(Items(i).ToString(), Items(i).ToString())
        Next i
        Return menu
    End Function

    Public Function CreateASPxComboBox() As ASPxComboBox
        Dim combo As New ASPxComboBox()
        combo.ID = "combo"
        AddHandler combo.CustomJSProperties, AddressOf Pass_CustomJSProperties
        Me.Controls.Add(combo)
        For i As Integer = 0 To Items.Count - 1
            combo.Items.Add(Items(i))
        Next i
        combo.EnableClientSideAPI = True
        combo.ClientSideEvents.SetEventHandler("SelectedIndexChanged", "OnSelectedIndexChanged")
        combo.ClientSideEvents.SetEventHandler("Init", "OnComboInit")
        Return combo

    End Function

    <System.ComponentModel.DefaultValue(RequestMethod.POST)> _
    Public Property RequestMethod() As RequestMethod

    <System.ComponentModel.DefaultValue(CustomControlType.ASPxComboBox)> _
    Public Property ControlType() As CustomControlType

    Protected Items As List(Of String)
    Public Sub PopulateItems()
        Items = ThemesProvider.GetThemes()
    End Sub

End Class